### PR TITLE
New deps

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,9 @@
     "async": "^2.5.0",
     "aug": "^2.0.0",
     "confi": "^9.0.0",
-    "lodash": "^4.11.1"
+    "deptree": "^1.0.0",
+    "lodash": "^4.11.1",
+    "vision": "^5.3.0"
```
Comment (jgallen23) [inline package.json]: why did vision get added a dep?

File: lib/plugins.js [hunk 2]
```diff
@@ -1,4 +1,6 @@
 const _ = require('lodash');
```
Comment (jgallen23) [inline lib/plugins.js]: is lodash still used?

File: lib/plugins.js [hunk 3]
```diff
@@ -18,7 +20,27 @@ module.exports = async (server, config, log, requireCwd) => {
...
     pluginArr.push(value);
   });
-  pluginArr = _.sortBy(pluginArr, '_priority');
+  // use deptree to resolve the order in which the plugins load
+  // based on their _dependencies:
+  const deptree = new Deptree();
+  pluginArr.forEach((plugin) => {
+    // priority is deprecated:
+    if (plugin._priority) {
+      console.log(`WARNING: '_priority' field used by ${plugin._name} is deprecated, please migrate your plugins to use the _dependencies format`);
```
Comment (dawnerd) [inline lib/plugins.js]: Lets change this to a server.log if we can

File: test/server-test.js [hunk 4]
```diff
@@ -44,3 +44,21 @@ lab.test('test server can load vision view engine ', async() => {
...
+});
+
+lab.test('notifies if deprecated _priority field still used', async() => {
+  const oldLog = console.log;
+  const results = [];
+  console.log = (input) => {
+    results.push(input);
+  };
+  const { server } = await hapiconfi(Hapi, { configPath: `${__dirname}/deprecated` });
+  console.log = oldLog;
```
Comment (dawnerd) [inline test/server-test.js]: Will have to change this to bind to the log event if changed to server.log

Comment (dawnerd) [reply test/server-test.js]: Yea actually throwing an error might just be the better way since it is kind of a breaking change.

**Pull request merged**